Fix SS host string substitution for SS host names

sheldon-b opened the pull request:
Not all endpoints use the 'us-3.rightscale.com' style host. This commit
removes the assumption that there will be a 'us' substring in the login
host. Fixes #45.

raphael commented:
Won't that break it for environments that have a specific shard suffix? (i.e. we need `selfservice-3` and not `selfservice`). Also consider using generic descriptions for these PRs, this project is public :)

sheldon-b commented:
It shouldn't as we've already split the host by hyphens and are only modifying one (the second from last) of those elements:
`elems := strings.Split(hostPrefix, "-")`
This is also the type of substitution we do in the SS module:
https://github.com/rightscale/rsc/blob/c1d7a4705d7b647798fd175858baf1ab8fbd45dd/ss/ss.go#L54

raphael commented:
Makes sense, merged. Thanks!
